frame_semantic_transformer/FrameSemanticTransformer.py

```diff
@@ -64,20 +67,67 @@ def __init__(
         batch_size: int = 8,
         predictions_per_sample: int = 5,
         inference_loader: Optional[InferenceLoader] = None,
+        logging_level: int = logging.WARNING,
     ):
+        caller_supplied_logging_level = logging_level
+        if logging_level not in logging._levelToName:
+            logging_level = logging.WARNING
+
+        logging.basicConfig(format="%(levelname)s: %(name)s: %(funcName)s: %(message)s")
```

Comment:
I think this will overwrite the logger for everything running in Python, not just for the logger for FrameSemanticTransformer. Could this line be removed? It would be unexpected for the user that creating a `FrameSemanticTransformer` instance would change the logging format for their whole application.

Reply:
Ah, sorry about that. I added another commit to fix this so that the formatter is configured directly for the `FrameSemanticTransformer` class logger, without touching the root logger.

Reply:
Why do we need to set the logger format here at all, though? It seems like it should be up to the user how they want their logs formatted?

Reply:
If a user has a preference for how their logs are formatted and configures the root logger at the start of their application, then the code I wrote respects that. The logger format doesn't strictly need to be set here in the case when the format hasn't been previously set, since there is a default. But the default format is just the bare message without the level information (e.g., WARNING) or where the message came from (e.g., the FrameSemanticTransformer constructor). If printing the bare message in the absence of user configuration is the preferred behavior, then I don't mind at all if you delete lines 78-85.

Reply:
👍  I merged this and removed the lines setting the logger. I just felt like it could be unexpected if the default log format changes for the whole app. Otherwise this looks great! Thank you for contributing this!